st2actions/st2actions/runners/mistral/v2.py

```diff
@@ -144,7 +144,10 @@ def try_run(self, action_parameters):
         inputs = self.runner_parameters.get('context', dict())
         inputs.update(action_parameters)
 
-        endpoint = 'http://%s:%s/v1/actionexecutions' % (cfg.CONF.api.host, cfg.CONF.api.port)
+        # If host value does not start with http, then default to http://.
+        protocol = 'http' if not cfg.CONF.api.use_ssl else 'https'
```

Comment:
I know I'm a bit late to the game, but `auth.api_url` config option was added in the past for this purpose. This setting contains a base public URL to the API endpoint which should be uses by all the clients (Mistral is also just another client).